### ISSUE TITLE
如何获得合成的异常图片和mask

### ISSUE THREAD

**Issue by Fantastic1115** [cqylunlun/GLASS#38]
您好，我想得到LAS中合成的异常图片和与之对应的Mask我应该怎么做呢？

**Comment by cqylunlun** [comment]
请对照mvtec.py中的[L203~L210](https://github.com/cqylunlun/GLASS/blob/af45306c72513409a72bb820d796e15d4b83e083/datasets/mvtec.py#L203-L210)，在适当位置，保存合成异常图像aug_image和同尺寸对应掩码mask_l。

**Comment by Fantastic1115** [comment]
您好，我修改了mvtec.py中的代码，但是保存的图片好像有点问题，你可以帮我看一下是哪里出错了吗？
    def __getitem__(self, idx):
        classname, anomaly, image_path, mask_path = self.data_to_iterate[idx]
        image = PIL.Image.open(image_path).convert("RGB")
        image = self.transform_img(image)

        mask_fg = mask_s = aug_image = torch.tensor([1])
        if self.split == DatasetSplit.TRAIN:
            aug = PIL.Image.open(np.random.choice(self.anomaly_source_paths)).convert("RGB")
            if self.rand_aug:
                transform_aug = self.rand_augmenter()
                aug = transform_aug(aug)
            else:
                aug = self.transform_img(aug)

            if self.class_fg:
                fgmask_path = image_path.split(classname)[0] + 'fg_mask/' + classname + '/' + os.path.split(image_path)[
                    -1]
                mask_fg = PIL.Image.open(fgmask_path)
                mask_fg = torch.ceil(self.transform_mask(mask_fg)[0])

            mask_all = perlin_mask(image.shape, self.imgsize // 8, 0, 6, mask_fg, 1)
            mask_s = torch.from_numpy(mask_all[0])
            mask_l = torch.from_numpy(mask_all[1])

            beta = np.random.normal(loc=self.mean, scale=self.std)
            beta = np.clip(beta, .2, .8)
            aug_image = image * (1 - mask_l) + (1 - beta) * aug * mask_l + beta * image * mask_l

            # 指定两个保存目录
            aug_image_dir = "/home/hxl/code/GLASS-main/ano/aug/"
            mask_dir = "/home/hxl/code/GLASS-main/ano/mask/"  
            os.makedirs(aug_image_dir, exist_ok=True)
            os.makedirs(mask_dir, exist_ok=True)

            # 保存合成图片
            aug_image_path = os.path.join(aug_image_dir, f"aug_image_{idx}.png")
            save_image(aug_image, aug_image_path)

            # 保存掩码
            mask_l_path = os.path.join(mask_dir, f"mask_l_{idx}.png")
            save_image(mask_l.unsqueeze(0), mask_l_path)

        if self.split == DatasetSplit.TEST and mask_path is not None:
            mask_gt = PIL.Image.open(mask_path).convert('L')
            mask_gt = self.transform_mask(mask_gt)
        else:
            mask_gt = torch.zeros([1, *image.size()[1:]])

        return {
            "image": image,
            "aug": aug_image,
            "mask_s": mask_s,
            "mask_gt": mask_gt,
            "is_anomaly": int(anomaly != "good"),
            "image_path": image_path,
        }
![34ffdf34286b5dc2f683b19a131a77c](https://github.com/user-attachments/assets/f1294d9a-b0a4-4e46-9c82-9e976144ade9)

**Comment by cqylunlun** [comment]
从图像来看，您没有进行反归一化操作且像素值尺度不正确。建议通过IMAGENET_MEAN和IMAGENET_STD将aug_image反归一化（到0~1之间），并乘以255（将像素值调整到0~255后），通过cv2保存。
